xtc/src/ledger.rs

```diff
@@ -100,17 +101,18 @@ async fn transfer(args: TransferArguments) -> Result<TransactionId, TransferErro
     let user = caller();
     crate::progress().await;
 
+    let fee = compute_fee(args.amount);
     let ledger = storage::get_mut::<Ledger>();
 
     ledger
-        .withdraw(&user, args.amount)
+        .withdraw(&user, args.amount + fee)
```

Comment:
Is there a particular reason why the withdraw doesn't return the 'TransferError::InsufficientBalance'? I see this .map_err(|_| TransferError::InsufficientBalance) in almost all places when we call the withdraw method.

Reply:
Yes, because `TransferError` is not always what we return, look at cycles_wallet.rs file, there we want to have send error as a string.

But we should refactor it to send `WithdrawError` and take overflows into account in that level.